File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/keys/IKeys.java
package org.ngbp.jsonrpc4jtestharness.rpc.keys;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.filterCodes.model.GetFilterCodes;
import org.ngbp.jsonrpc4jtestharness.rpc.keys.model.Keys;

@JsonRpcType
public interface IKeys {
    @JsonRpcMethod("org.atsc.request.keys")
    Keys requestKeys();

    @JsonRpcMethod("org.atsc.relinquish.keys")
    Object relinquishKeys();

    @JsonRpcMethod("org.atsc.notify")
    Object requestKeysTimeout();
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/keys/IKeys.kt
package org.ngbp.jsonrpc4jtestharness.rpc.keys

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcParam
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.RpcResponse
import org.ngbp.jsonrpc4jtestharness.rpc.keys.model.Keys

@JsonRpcType
interface IKeys {
    @JsonRpcMethod("org.atsc.request.keys")
    fun requestKeys(@JsonRpcParam("keys") listOfKeys: List<String>): Keys

    @JsonRpcMethod("org.atsc.relinquish.keys")
    fun relinquishKeys(@JsonRpcParam("keys") listOfKeys: List<String>): RpcResponse

    @JsonRpcMethod("org.atsc.notify")
    fun requestKeysTimeout(): RpcResponse
}